snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java

```diff
@@ -415,6 +416,17 @@ public void initialize() throws OperatorException {
         }
 
         setAutoGrouping();
+
+        // Add heading angles
+        final MetadataElement slavesHeadingAnglesElem = new MetadataElement("Slaves_Heading_Angles");
+        final MetadataElement abstractedMetadataTarget = targetProduct.getMetadataRoot().getElement("Abstracted_Metadata");
```

Comment:
@estebanag collocate is primarily an optical operator and optical products may not have an Abstracted_Metadata element. This needs to check if the element exists

Reply:
Yes, it should be checked if targetProduct and slaveProducts have Abstracted_Metadata and only then the heading elements should be added. If this is done then this is okay for me.

Reply:
I just updated it. Is this what you had in mind?

Reply:
Looks good. Have you tested with optical products?

Reply:
Just ran it with Sentinel-2 data and it went well. I also debugged it, and the check for `Abstracted_Metadata` works as expected.